BootloaderCorePkg/Library/FspApiLib/FspSiliconInit.c

```diff
@@ -36,24 +37,26 @@ CallFspSiliconInit (
 
   ASSERT (FspHeader->Signature == FSP_INFO_HEADER_SIGNATURE);
   ASSERT (FspHeader->ImageBase == PcdGet32 (PcdFSPSBase));
+  FspsUpdptr = AllocatePool (FixedPcdGet32 (PcdFSPSUpdSize));
 
+  Status = PcdSet32S (PcdFspsUpdPtr,(UINT32)(UINTN)FspsUpdptr);
   // Copy default UPD data
   DefaultSiliconInitUpd = (UINT8 *)(UINTN)(FspHeader->ImageBase + FspHeader->CfgRegionOffset);
-  CopyMem (&FspsUpd, DefaultSiliconInitUpd, FspHeader->CfgRegionSize);
+  CopyMem (FspsUpdptr, DefaultSiliconInitUpd, FspHeader->CfgRegionSize);

```

Comment:
you need to ensure there is no buffer overflow.   you should only copy up to the buffer size elements.

Reply:
I think we could remove PcdFSPSUpdSize to use FspHeader->CfgRegionSize when allocate FSPS UPD memory.